### PR TITLE
bluetooth: fast_pair: Cleanup Fast Pair unit tests

### PR DESCRIPTION
**Pull request by MarekPieta**
The fp_crypto and fp_storage are built as a separate libraries to clean up code integration in the unit tests.

Jira: NCSDK-15610

### CODE REVIEW

**Review by alstrzebonski — APPROVED** [review]

File: tests/subsys/bluetooth/fast_pair/crypto/Kconfig [hunk 1]
```diff
@@ -4,27 +4,10 @@
...
-	default BT_FAST_PAIR_CRYPTO_MBEDTLS
-
-config BT_FAST_PAIR_CRYPTO_TINYCRYPT
-	bool "Fast Pair with Tinycrypt cryptographic backend"
-	select TINYCRYPT
-	select TINYCRYPT_SHA256
-	select TINYCRYPT_AES
-	select TINYCRYPT_ECC_DH
+menu "Test configuration"
+rsource "../../../../../subsys/bluetooth/services/fast_pair/fp_crypto/Kconfig.fp_crypto"
```
Comment (KAGA164) [inline tests/subsys/bluetooth/fast_pair/crypto/Kconfig:8]: Maybe you could use here an absolute path?
Starting with `source "${NRF_DIR}/`

File: subsys/bluetooth/services/fast_pair/fp_storage/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+target_include_directories(fp_storage PUBLIC include)
+target_include_directories(fp_storage PUBLIC ../include/common)
+if(CONFIG_BT_FAST_PAIR_STORAGE_EXPOSE_PRIV_API)
+  target_include_directories(fp_storage PUBLIC include_priv)
+else()
+  target_include_directories(fp_storage PRIVATE include_priv)
+endif()
+
+target_link_libraries(fp_storage PRIVATE zephyr_interface)
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply subsys/bluetooth/services/fast_pair/fp_storage/CMakeLists.txt:21]: See comment above

File: tests/subsys/bluetooth/fast_pair/crypto/Kconfig (hunk 1 above)
Comment (MarekPieta) [reply tests/subsys/bluetooth/fast_pair/crypto/Kconfig:8]: It does not seem to work. That's why I used the relative path here (assuming it's good enough for unit test usecase)

File: tests/subsys/bluetooth/fast_pair/storage/Kconfig [hunk 3]
```diff
@@ -4,18 +4,10 @@
...
-	int "Maximum number of stored Account Keys"
-	range 5 10
-	default 5
-	help
-	  Maximum number of stored account keys. According to Fast Pair specification, it should be
-	  at least equal to 5. It can not be larger than 10, because then Account Key Filter length
-	  would not fit in the "field length and type" data field specified in the non-discoverable
-	  advertising packet.
+menu "Test configuration"
+rsource "../../../../../subsys/bluetooth/services/fast_pair/fp_storage/Kconfig.fp_storage"
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply tests/subsys/bluetooth/fast_pair/storage/Kconfig:8]: I did not see any, but maybe I missed something

**Review by rakons — APPROVED** [review]

File: subsys/bluetooth/services/fast_pair/fp_crypto/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+add_library(fp_crypto STATIC)
+
+if(CONFIG_BT_FAST_PAIR_CRYPTO)
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply subsys/bluetooth/services/fast_pair/fp_crypto/CMakeLists.txt:9]: Maybe you should add this subdirectory only when this option is enabled?

File: subsys/bluetooth/services/fast_pair/fp_storage/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+add_library(fp_storage STATIC)
+
+if(CONFIG_BT_FAST_PAIR_STORAGE)
```
Comment (pdunaj) [inline subsys/bluetooth/services/fast_pair/fp_storage/CMakeLists.txt:9]: add this cmake file only if `CONFIG_BT_FAST_PAIR_STORAGE` is enabled.

**Review by pdunaj — APPROVED** [review]

File: tests/subsys/bluetooth/fast_pair/crypto/Kconfig (hunk 1 above)
Comment (MarekPieta) [reply tests/subsys/bluetooth/fast_pair/crypto/Kconfig]: Fixed with help of @rakons

File: tests/subsys/bluetooth/fast_pair/storage/Kconfig (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply tests/subsys/bluetooth/fast_pair/storage/Kconfig]: Also fixed with help of @rakons

**Comment by MarekPieta** [comment]
@KAGA164 and @kapi-no are on vacation and we already have a few PRs waiting for these changes. Can we merge it?
I applied most of the changes requested by @KAGA164  (I only did not make it as zephyr_library, because I prefer to keep it as a small private library)

**Pull request merged**